pyproject.toml

```
@@ -43,8 +43,15 @@ dependencies = [
     'pillow (>=10.0.0,<12.0.0)',
     'pyyaml (>=5.1,<7.0.0)',
     'typing-extensions (>=4.12.2,<5.0.0)',
-    'typer (>=0.12.5,<0.20.0)',
+    'typer (>=0.12.5,<0.17.0)',
     'latex2mathml (>=3.77.0,<4.0.0)',
+    "tree-sitter==0.23.2",
+    "tree-sitter-python==0.23.6",
+    "tree-sitter-c==0.23.4",
+    "tree-sitter-java==0.23.5",
+    "tree-sitter-javascript==0.23.1",
+    "tree-sitter-typescript==0.23.2",
+
```

In pyproject.toml we should provide a range for the versions (otherwise we risk dependency hell, plus users get no upstream updates).
Unless there are specific reasons for different ranges, one could go with this:
```suggestion
    'typer (>=0.12.5,<0.20.0)',
    'latex2mathml (>=3.77.0,<4.0.0)',
    "tree-sitter (>=0.23.2,<1.0.0)",
    "tree-sitter-python (>=0.23.6,<1.0.0)",
    "tree-sitter-c (>=0.23.4,<1.0.0)",
    "tree-sitter-java (>=0.23.5,<1.0.0)",
    "tree-sitter-javascript (>=0.23.1,<1.0.0)",
    "tree-sitter-typescript (>=0.23.2,<1.0.0)",
```